### PR TITLE
Add initial device support for Sony Xperia T3 (sony-seagull)

### PR DESCRIPTION
**Pull request by MayeulC**
The kernel is fetched from a sony copyleft archive, version 3.4.0.
Display, touch and USB networking are working.

In essence, this is reopening #1180 
@PureTryOut convinced me to merge this as a starter, it only needed to be updated to the latest devicenfo format. I might still try to port a more recent kernel, but I don't have much time right now...

I am not sure I can mirror the archive on GitHub. It's 142MB, and doesn't fit in a "release" post. Will it fit in a depot?

### CODE REVIEW

**Comment by MayeulC** [comment]
Wow, there are some new fancy wiki checks!
Not sure why the CI failed, though...

**Comment by ollieparanoid** [comment]
Thanks for making the PR! I didn't review it yet, but for here's a mirror of the source code:
https://github.com/pmsourcedump/android_kernel_sony_seagull
Also I'm not sure why Travis failed either (possibly the log was too long?) - I've kicked it off again.

**Comment by mirh** [comment]
Ehrm.. Didn't the tree I linked tickled?

File: aports/device/device-sony-seagull/APKBUILD [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Reference: <https://postmarketos.org/devicepkg>
+pkgname="device-sony-seagull"
+pkgdesc="Sony Xperia T3"
+pkgver=0.1
+pkgrel=0
+url="https://postmarketos.org"
+license="MIT"
+arch="noarch"
+options="!check"
+depends="linux-sony-seagull mdss-fb-init-hack mkbootimg mesa-dri-swrast"
```
Comment (ollieparanoid) [inline aports/device/device-sony-seagull/APKBUILD]: [This is why Travis failed](https://travis-ci.org/postmarketOS/pmbootstrap/builds/361777460#L4906) (in the second run, the log did not get cut off):
> Missing 'postmarketos-base' in depends of /home/travis/build/postmarketOS/pmbootstrap/aports/device/device-sony-seagull/APKBUILD

File: aports/device/linux-sony-seagull/APKBUILD [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+url="https://kernel.org"
+license="GPL2"
+options="!strip !check !tracedeps"
+makedepends="perl sed installkernel bash gmp-dev bc linux-headers elfutils-dev dtbtool"
+HOSTCC="${CC:-gcc}"
+HOSTCC="${HOSTCC#${CROSS_COMPILE}}"
+
+# Source
+_archive="18.1.A.1.23.tar.bz2"
+_srcurl="https://dl.developer.sonymobile.com/code/copylefts/${_archive}"
```
Comment (ollieparanoid) [inline aports/device/linux-sony-seagull/APKBUILD]: The kernel of that big archive is mirrored to https://github.com/pmsourcedump/android_kernel_sony_seagull now.

**Comment by mirh** [comment]
I didn't link some random whatever kernel..
https://github.com/sonyxperiadev/kernel/tree/aosp/LA.BF64.1.2.2_rb4.7 

In other news I found Yukon dts even on this quite newer branch https://github.com/sonyxperiadev/kernel/tree/aosp/LA.BR.1.3.3_rb2.14

**Comment by MayeulC** [comment]
@mirh No, I didn't use this kernel for now. I looked into it, and even mainlining the device, but I was kind of busy lately, and I think that merging it with the vendor fork is better than not merging it at all.
I am still planning to work more on this.

@ollieparanoid: I don't know about using our own kernel repository. On one hand, people who use postmarketOS already rely on us for security, and "controlling" (still hosted on GitHub) the repository is better. On the other hand, fetching it from the official source seems more legit (and since the hash is the same, the replacement is easily verifiable). I could add the backup as a comment, or use it directly, which is what you would prefer, if I got it right?

**Comment by mirh** [comment]
That's still downstream kernel tbh (so @opendata-like efforts would be needed to make "the big leap")
But the point would be anyway, that it's a way newer branch then what stock rom had.  3.10 instead of 3.4 is quite huge imo. 

Anyway, enough said, it's all your choice in the end.

**Comment by ollieparanoid** [comment]
@MayeulC: I would prefer downloading the source from our own mirror, so the download is smaller.

**Comment by MayeulC** [comment]
So, I'm back on this one -- on and off, sorry about that, I don't have much time these days.
@ollieparanoid  Actually, I had another mirror [here](https://github.com/MayeulC/sony-seagull-kernel/commit/a24716704fed4d60792944ec55565791e8aa4001) (I should add tags), but thought you wanted to retain the checksum compatibility. As for a big kernel bump, I am afraid this will have to wait.

**Pull request merged**